libs/auth-api-lib/src/lib/user-identities/user-identities.service.ts

```diff
@@ -214,4 +217,30 @@ export class UserIdentitiesService {
 
     return null
   }
+
+  async updateClaims(
+    subjectId: string,
+    claims: ClaimDto[],
+  ): Promise<ClaimDto[]> {
+    await this.sequelize.transaction(async (t) => {
+      await this.claimModel.destroy({
+        where: { subjectId: subjectId },
+        transaction: t,
+      })
+
+      await this.claimModel.bulkCreate(
+        claims.map((c) => ({
+          subjectId: subjectId,
+          type: c.type,
+          value: c.value,
+          valueType: c.valueType,
+          issuer: c.issuer,
+          originalIssuer: c.originalIssuer,
+        })),
+        { transaction: t },
+      )
+    })
+
+    return this.claimModel.findAll({ where: { subjectId: subjectId } })
```

Comment:
As we are writing and reading in the same service call we should add `useMaster: true`, we @eirikurn  saw issue when creating delegations as the query read from read replica which was not updated.

Reply:
Good point, fixed.